src/Utils/Rotation.cpp

```diff
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
 #ifndef SKIP_ASSERT
     double sqrtNorm = static_cast<double>(this->squaredNorm());
     bool checkUnit(fabs(sqrtNorm - 3.) < 1e-4);
-    if (!checkUnit){
-        // Make calulation here
-        utils::Matrix3d mat;
+    if (!checkUnit)
+    {
+
+        // Code addapted from https://github.com/brainexcerpts/3x3_polar_decomposition/blob/master/src/polar_decomposition_3x3.inl
+
+        // Placer inf_norm and one norm in a function
+        // créer fonction polar_decomposition et déplacer dans Matrix3d
+
+        float tolerance = 1e-6;
+        utils::Matrix3d M = this->block(0, 0, 3, 3);
+        utils::Matrix3d R;
+        utils::Matrix3d S;
+        utils::Matrix3d Mk;
+        utils::Matrix3d Ek;
+        float det, E_one_norm;
+        bool use_svd = false;
+
+        
+        // Mk = mat^T
+        Mk = M.transpose();
+        
+        //M one-norm
+        float M_one_norm = 0.0f;
+        for (int i = 0; i < 3; i++)
+        {
+            float col_abs_sum = fabs(M(i, 0)) + fabs(M(i, 1)) + fabs(M(i, 2));
+            if (col_abs_sum > M_one_norm)
+                M_one_norm = col_abs_sum;
+        }
+
+        //M infinity-norm
+        float M_inf_norm = 0.0;
+        for (int i = 0; i < 3; i++)
+        {
+            float row_sum = fabs(M(i, 0)) + fabs(M(i, 1)) + fabs(M(i, 2));
+            if (row_sum > M_inf_norm)
+                M_inf_norm = row_sum;
+        }
```

Comment:
I think this for loop does the exact same as the one before? Did you mean M(0, i) in this one or the previous?
Also, since they both require i = 0 to 2, they probably can be merge into one for loop